lib/ibm_speech_to_text/sink.ex

```diff
@@ -1,23 +1,110 @@
+  alias Membrane.Buffer
+  alias Membrane.Caps.Audio.FLAC
+  alias Membrane.Event.EndOfStream
+  alias Membrane.Time
+  alias IBMSpeechToText.{Client, Message, Response}
 
-  def_input_pads input: [
-    caps: :any,
-    demand_unit: :buffers
-  ]
+  def_input_pad :input, caps: FLAC, demand_unit: :buffers
+
+  def_options region: [
+                description: """
+                Region in which the endpoint is located.
+                See `t:IBMSpeechToText.region/0`
+                """,
+                type: :atom,
+                spec: IBMSpeechToText.region()
+              ],
+              api_key: [
+                description: """
+                API key for the Speech to Text Service
+                """,
+                type: :string
+              ]
+
+  @impl true
+  def handle_init(opts) do
+    state =
+      opts
+      |> Map.from_struct()
+      |> Map.merge(%{
+        start_time: nil,
+        samples: 0,
+        sample_rate: 16_000,
+        connection: nil
+      })
+
+    {:ok, state}
+  end
 
   @impl true
   def handle_prepared_to_playing(_context, state) do
-    {{:ok, demand: :input}, state}
+    with {:ok, pid} <- Client.start_link(state.region, state.api_key) do
+      {{:ok, demand: :input}, %{state | connection: pid}}
+    end
+  end
+
+  @impl true
+  def handle_caps(:input, %FLAC{} = caps, _, %{connection: conn} = state) do
+    Client.send_message(conn, %Message.Start{content_type: :flac, interim_results: true})
+    start_time = Time.os_time()
+    state = %{state | start_time: start_time, sample_rate: caps.sample_rate}
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_event(:input, %EndOfStream{}, _, %{connection: conn} = state) do
+    Client.send_message(conn, %Message.Stop{})
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_event(pad, event, ctx, state) do
+    super(pad, event, ctx, state)
   end
 
   @impl true
-  def handle_demand(_pad, _size, _unit, _ctx, state) do
+  def handle_write(
+        :input,
+        %Buffer{payload: payload, metadata: %FLAC.FrameMetadata{} = meta},
+        _ctx,
+        %{connection: conn, sample_rate: sample_rate, start_time: start_time} = state
+      ) do
+    Client.send_data(conn, payload)
+
+    next_sample_num = meta.starting_sample_number + meta.samples
+    next_frame_time = start_time + trunc(next_sample_num * Time.seconds(1) / sample_rate)
+    demand_time = (next_frame_time - Time.os_time()) |> max(0) |> Time.to_milliseconds()
+    Process.send_after(self(), :demand_frame, demand_time)
+
     {:ok, state}
   end
 
   @impl true
-  def handle_write(_pad, _payload, _ctx, state) do
+  def handle_write(:input, %Buffer{payload: payload}, _ctx, %{connection: conn} = state) do
+    Client.send_data(conn, payload)
+    {{:ok, demand: :input}, state}
+  end
+
+  @impl true
+  def handle_other(%Response{} = response, _ctx, state) do
+    transcipts =
+      response.results
+      |> Enum.filter(fn result -> result.final end)
```

Comment:
do this filter in the lib

Reply:
Nope, as someone might need the interim results (not final)